engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobFailProcessor.java

```diff
@@ -82,12 +91,26 @@ public void afterAccept(
   }
 
   private void acceptCommand(
-      final TypedRecord<JobRecord> command, final CommandControl<JobRecord> commandControl) {
+      final TypedRecord<JobRecord> command,
+      final CommandControl<JobRecord> commandControl,
+      final Consumer<SideEffectProducer> sideEffect) {
     final long key = command.getKey();
     final JobRecord failedJob = jobState.getJob(key);
-    failedJob.setRetries(command.getValue().getRetries());
+    final var retries = command.getValue().getRetries();
+    final var retryBackOff = command.getValue().getRetryBackOff();
+    failedJob.setRetries(retries);
     failedJob.setErrorMessage(command.getValue().getErrorMessageBuffer());
+    failedJob.setRetryBackoff(retryBackOff);
     commandControl.accept(JobIntent.FAILED, failedJob);
     jobMetrics.jobFailed(failedJob.getType());
+    if (retries > 0 && retryBackOff > 0) {
+      sideEffect.accept(
+          () -> {
+            final long receivedTime = command.getTimestamp();
+            failedJob.setRecurringTime(receivedTime + retryBackOff);
```

Comment:
:x: We need to set the `recurringTime` before writing the event by calling `commandControl.accept(JobIntent.FAILED, failedJob);`.

The job event `failed` is written by calling `commandControl.accept(JobIntent.FAILED, failedJob);`. The side effects are executed after the event is written. If the `recurringTime` is set in the side effects then it is not serialized in the event and doesn't have any effect.

I guess that you wanted to avoid duplicating the condition and I like it :+1: 
We could move the setting of the `recurringTime` out of the side effects and move the two lines L104-105 (i.e. `commandControl.accept(JobIntent.FAILED, failedJob);    jobMetrics.jobFailed(failedJob.getType());`) below the `if` block.

Reply:
Ooops, this is a foolish mistake of mine 😅